### DISCUSSION TITLE
Fail to build Zephyr when enabling STM32 lis2dw12 accelerometer driver

### DISCUSSION THREAD

**Discussion by gineelee** [zephyrproject-rtos/zephyr#55151] [Q&A]
It seems two files lis2dw12_reg.c and lis2dw12_reg.h are missing in Zephyr source tree.
https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers/sensor/lis2dw12

I can successfully build the code after manually add these two files from https://github.com/STMicroelectronics/lis2dw12-pid.

Is it supposed to do this way or something I miss?

**Comment by yashi** [comment]
`lis2dw12_reg.c` and `lis2dw12_reg.h` are provided by [`hal_st`](https://github.com/zephyrproject-rtos/hal_st). See https://github.com/zephyrproject-rtos/hal_st/tree/master/sensor/stmemsc/lis2dw12_STdC/driver

If you follow [Application Development](https://docs.zephyrproject.org/latest/develop/application/index.html#application-development), the hal should be available to your app.

If you are on your own, check your `west.yml` to see `hal_st` is there.  Please note that the `hal_st` can be imported from Zephyr repo, just like https://github.com/zephyrproject-rtos/example-application/blob/main/west.yml does.
